### ISSUE TITLE
Doesn't work with monorepos

### ISSUE THREAD

**Issue by dmtrKovalenko** [abhinaba-ghosh/cypress-react-selector#55]
This line completely breaks  monorepo setup. In case node_modules are hosted somewhere upper. Please replace this with `require`. 

https://github.com/abhinaba-ghosh/cypress-react-selector/blob/62e8da928d28fd0c3a985da4ee9604daf983f404/src/resqInjector.js#L9

**Comment by abhinaba-ghosh** [comment]
Hi @dmtrKovalenko , Thanks for addressing this issue. Just released the version `2.0.5`. Please try and let us know if that resolves the issue.

**Comment by abhinaba-ghosh** [comment]
Re-opening as there `require.resolve` is erroring out in the bundle. Unpublished the `2.0.5` version for a while. Will release it later today.

**Comment by abhinaba-ghosh** [comment]
should work with `v2.0.6`

**Comment by abhinaba-ghosh** [comment]
Hi @dmtrKovalenko , after a series of trial and error, I understood that providing a native support for this request is not going to be possible any time soon. The behind is cypress only allows basic node operations in the plugin file. But, you can overwrite the existing functionality of `waitForReact`. 


in cypress/plugins/index.js 

```js
// declare task
const fs = require('fs');

module.exports = (on, config) => {
  on('task', {
    readResqFile() {
      if (fs.existsSync(require.resolve('resq'))) {
        return fs.readFileSync(require.resolve('resq'), 'utf8');
      }
      return null;
    },
  });
};

```

Now create your own copy of `waitForReact`

in cypress/support/index.js

```js
const waitForReact = (timeout = 10000, reactRoot) => {
  cy.task('readResqFile').then((script) => {
    cy.window({ log: false }).then({ timeout: timeout }, (win) => {
      win.eval(script);
      return new Cypress.Promise.resolve(
        win.resq.waitToLoadReact(timeout, reactRoot)
      )
        .then(() => {
          cy.log('[cypress-react-selector] loaded');
        })
        .catch((err) => {
          throw new Error(
            `[cypress-react-selector] root found as ${reactRoot}. It is not valid root for your application`
          );
        });
    });
  });
};

Cypress.Commands.add('waitForReact', waitForReact);
```

Now, you can proceed with your own copy and use the other functionalities of `cypress-react-selector` as usual.

**Comment by nickpalmer** [comment]
It isn't the require.resolve that is the problem it is the use of fs.existsSync.

**Comment by abhinaba-ghosh** [comment]
Thanks, @nickpalmer for pointing and fixing this issue. The fix is available in `v2.2.1`.
